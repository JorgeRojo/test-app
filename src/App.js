import React from 'react';
import UserList from './UserList';

const App = ({ title }) => {
  return (
    <div className="App">
      <h1>{title || 'User list'}</h1>
      <UserList />
    </div>
  );
};

export default App;
