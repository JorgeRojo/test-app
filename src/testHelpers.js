import React from 'react';

export const mockComponent = (name, { children, ...props }) => (
  <code name={name} props={JSON.stringify(props, null, 2)}>
    {children}
  </code>
);
