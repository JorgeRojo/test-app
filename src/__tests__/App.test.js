import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { mockComponent } from '../testHelpers';
import App from '../App';

jest.mock('../UserList', () => props => mockComponent('UserList', props));

describe('App.js', () => {
  let container = null;

  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('renders without crashing', () => {
    render(<App />, container);
    unmountComponentAtNode(container);
  });

  it('renders data changes', () => {
    act(() => {
      render(<App />, container);
    });
    expect(container).toMatchSnapshot();

    act(() => {
      render(<App title="this is my data" />, container);
    });
    expect(container).toMatchSnapshot();
  });
});
