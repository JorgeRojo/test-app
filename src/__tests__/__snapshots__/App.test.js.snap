// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js renders data changes 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      User list
    </h1>
    <code
      name="UserList"
      props="{}"
    />
  </div>
</div>
`;

exports[`App.js renders data changes 2`] = `
<div>
  <div
    class="App"
  >
    <h1>
      this is my data
    </h1>
    <code
      name="UserList"
      props="{}"
    />
  </div>
</div>
`;
